pulp_smash/tests/ostree/api_v2/test_sync_publish.py

```diff
@@ -0,0 +1,447 @@
+
+    def test_task_traceback(self):
+        """Assert each task's "traceback" field is non-null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertIsNotNone(task_body['traceback'])
+
+    def test_number_tasks(self):
+        """Assert that two task were spawned."""
+        self.assertEqual(len(self.task_bodies), 1)
+
+
+class SyncInvalidFeedTestCase(_BaseTestCase):
+    """With invalid feed or branch, the sync completes with reported errors."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create an OSTree repository with an invalid feed or branch, sync."""
+        super(SyncInvalidFeedTestCase, cls).setUpClass()
+        client = api.Client(cls.cfg, api.json_handler)
+        bodies = tuple(_gen_ostree_repo_body() for _ in range(2))
+        bodies[0]['importer_config']['feed'] = utils.uuid4()  # invalid feed
+        bodies[0]['importer_config']['branches'] = [_VALID_BRANCHES[0]]
+        bodies[1]['importer_config']['feed'] = _VALID_FEED
+        bodies[1]['importer_config']['branches'] = [utils.uuid4()]
+        repos = tuple(client.post(REPOSITORY_PATH, body) for body in bodies)
+        client.response_handler = api.echo_handler
+        cls.sync_reports = []
+        for repo in repos:
+            cls.sync_reports.append(client.post(
+                urljoin(repo['_href'], 'actions/sync/'),
+                {'override_config': {}}
+            ))
+            cls.sync_reports[-1].raise_for_status()
+        cls.task_bodies = tuple(chain.from_iterable(
+            utils.poll_spawned_tasks(cls.cfg, report.json())
+            for report in cls.sync_reports
+        ))
+        for repo in repos:
+            cls.resources.add(repo['_href'])
+
+    def test_start_sync_code(self):
+        """Assert the call to sync a repository returns an HTTP 202."""
+        for sync_repo in self.sync_reports:
+            with self.subTest(sync_repo=sync_repo):
+                self.assertEqual(sync_repo.status_code, 202)
+
+    def test_task_error(self):
+        """Assert each task's "error" field is non-null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertIsNotNone(task_body['error'])
+
+    def test_task_traceback(self):
+        """Assert each task's "traceback" field is non-null."""
+        for i, task_body in enumerate(self.task_bodies):
+            with self.subTest(i=i):
+                self.assertIsNotNone(task_body['traceback'])
+
+    def test_error_details(self):
+        """Assert that some task's progress report contains error details."""
+        for i, task_body in enumerate(self.task_bodies):
+            error_details_num = 0
+            for action in task_body['progress_report']['ostree_web_importer']:
+                if action['error_details'] != []:
+                    error_details_num += 1
+            self.assertNotEqual(error_details_num, 0)
```

Comment:
This test will tell us how many OSTree web importers have errors, but not which ones. Try something like this instead:

```python
for i, task_body in enumerate(self.task_bodies):
    for j, action in enumerate(task_body['progress_report']['ostree_web_importer']):
        with self.subTest((i, j)):
            self.assertEqual(action['error_details'], [])
```